### ISSUE TITLE
[Deliverable] Review usage of content topics in Status Communities protocol

### ISSUE THREAD

**Issue by chair28980** [waku-org/pm#268]
The usage of content topics in Status is aligned with Wakuv1. Waku v2 comes with a new recommended format that enables auto-sharding.

Moreover, single Status users currently use a high number of content topics, which may have an impact on performance of req-res protocols such as store and filter.

Such impact is to be measured in a previous milestone by Vac DST.

The output of this deliverable should be an RFC update on how content topics should be used, backed with simulations when performance improvement is expected.

It should include migration strategy and potential impact on the product.

**Comment by fryorcraken** [comment]
I see that the proposal is to simply apply the change across all communities.

This is the best way to get it across the board. However, it means that benefits for store queries (less content topics) will not be noticeable until `n+2`.

I would suggest to make it so that new communities from `n+1` only use single content topic, for read and write, so that they can benefit from it directly.

It would mean being able to identify if a community uses a single content topic or not. This may need to be coded in community description and/or community link/invite.

See https://forum.vac.dev/t/breaking-changes-and-roll-out-strategies/338/5 for reference.

cc @sunleos @iurimatias @ilmotta

**Comment by chaitanyaprem** [comment]
> I would suggest to make it so that new communities from `n+1` only use single content topic, for read and write, so that they can benefit from it directly.
> 
> It would mean being able to identify if a community uses a single content topic or not. This may need to be coded in community description and/or community link/invite.

Good Point, if we can somehow indicate this in community description that should suffice to merge both phases for new communities and rollout second phase later for existing communities. will discuss with @osmaczko on how to achieve this.

**Comment by chaitanyaprem** [comment]
> I would suggest to make it so that new communities from `n+1` only use single content topic, for read and write, so that they can benefit from it directly.

fyi, as per discussion [here](https://github.com/status-im/status-go/pull/5864#issuecomment-2490601821), looks like this is not preferred as it is complicating code.
hence sticking to original approach of 2 phased deployment for now.

**Comment by fryorcraken** [comment]
Lets close.
